@here/harp-mapview-decoder/lib/TileDataSource.ts

```diff
@@ -255,8 +255,17 @@ export class TileDataSource<TileType extends Tile> extends DataSource {
 
                 tile.setDecodedTile(tileLoader.decodedTile);
             } else {
-                // empty tiles are traditionally ignored and don't need decode
-                tile.forceHasGeometry(true);
+                if (tileLoader.decodedTile !== undefined) {
+                    // to render at least an empty surface
+                    // when the data is loaded
+                    // tile requires "decodedTile" object,
+                    tile.setDecodedTile(tileLoader.decodedTile);
+                }
```

Comment:
This to me sounds like a hack, we are giving the tile empty data to force it to render.
Please fix the root cause and not add a work around like this to "fix" the issue.

Reply:
A lot of logic depend on "decodedTile" property of the Tile class.
Including 
`tile.prepareForRender() {
    tile.createGeometries(decodedTile) {
        tile.createObjects(decodedTile, this.objects) {
            tile.createPlane(...);
        }
    }
 }`
call stack.
if the tile has no objects some decoders return empty decodedTile with no geometrys, and some of them return "unifuned".
the idea was to universalize behavior.
If the tile is not loadet the tile.decodedTile = undefined. Then when we get the data, but without any objects for the tile, then tile.decodedTile  = {default empty obj};
so behavior will be more predictable, and there will be less "if else" statements in the logic

Reply:
> the idea was to universalize behavior

This is good idea, but relying on `tile.decodedTile` is wrong because we want to forget `decodedTile` (and let GC kick-in) as soon as geometry is created.
This is the reason why `Tile.prepareForRender` resets `this.m_decodedTile` as soon geometry is created.

...

That's why we have `hasGeometry` flag.
This is definitive marker for rest of code, that this tile should be rendered.

On the other hand, w.r.t to you patch.
All this fiddling with flags can be replaced by:

Either this:
```
 tileLoader.loadAndDecode().then(() => {
     const decodedTile = tileLoader.decoderTile || defaultEmptyDecodedTile;
     // copyright stuff
     tile.setDecodedTile(decodedTile);
}
```
or this:

```
// Tile.forceHasGeometry
    forceHasGeometry() {
        this.createGeometries(defaultEmptyDecodedTile)
    }
```

Both of these approaches should fix issue.

Reply:
I like the idea that the forceHasGeometry calls createGeometry.... or maybe rather not call createGeometries, but some other simpler method, createDefaultPlane